assets/js/ship-builder.js

```diff
@@ -108,11 +109,24 @@ function drawImage() {
 	][swizzle];
 	const imageData = context.getImageData(0, 0, canvas.width, canvas.height);
 	const pixels = imageData.data;
+	let swizzlePixels = null;
+	if(swizzleMask) {
+		var swizzleCanvas = document.createElement('canvas');
+		var swizzleContext = swizzleCanvas.getContext("2d");
+		swizzleCanvas.width = canvas.width;
+		swizzleCanvas.height = canvas.height;
+		swizzleContext.drawImage(swizzleMask, 0, 0, canvas.width, canvas.height);
+		var swizzleData = swizzleContext.getImageData(0, 0, canvas.width, canvas.height);
```

Comment:
You don't need these to be `var`, do you? I don't think they are used outside of this scope.

Reply:
```suggestion
		const swizzleCanvas = document.createElement('canvas');
		const swizzleContext = swizzleCanvas.getContext("2d");
		swizzleCanvas.width = canvas.width;
		swizzleCanvas.height = canvas.height;
		swizzleContext.drawImage(swizzleMask, 0, 0, canvas.width, canvas.height);
		const swizzleData = swizzleContext.getImageData(0, 0, canvas.width, canvas.height);
```